schema-nft-marketplace.graphql

```diff
@@ -0,0 +1,109 @@
+# Subgraph Schema: NFT Marketplace
+# Version: 0.0.1
+
+enum Network {
+  ARBITRUM_ONE
+  ARWEAVE_MAINNET
+  AURORA
+  AVALANCHE
+  BOBA
+  BSC # aka BNB Chain
+  CELO
+  COSMOS
+  CRONOS
+  MAINNET # Ethereum Mainnet
+  FANTOM
+  FUSE
+  HARMONY
+  JUNO
+  MOONBEAM
+  MOONRIVER
+  NEAR_MAINNET
+  OPTIMISM
+  OSMOSIS
+  MATIC # aka Polygon
+  XDAI # aka Gnosis Chain
+}
+
+enum NftStandard {
+    ERC721
+    ERC721A
+    ERC1155
+}
+
+enum SaleStrategy {
+    " Strategy that executes an order at a fixed price that can be taken either by a bid or an ask. "
+    StandardSaleForFixedPrice
+    " Strategy that executes an order at a fixed price that can be matched by any tokenId for the collection. "
+    AnyItemFromCollectionForFixedPrice
+    " Strategy that executes an order at a fixed price that can be matched by any tokenId in a set of tokenIds. "
+    AnyItemInASetForFixedPrice
+    " Strategy to launch a Dutch Auction for a token where the price decreases linearly until a specified timestamp and end price defined by the seller. "
+    DutchAuction
+    " Strategy to set up an order that can only be executed by a specific address. "
+    PrivateSale
+}
+
+type Marketplace @entity {
+    id: ID!
+    name: String!
+    network: Network!
+    collectionCount: Int!
+    " Cumulative sale volume (in ETH) "
+    cumulativeSaleVolumeETH: BigDecimal!
```

Comment:
I think we should accumulate volume per token/currency, this way we don't need to worry about USD conversion. Otherwise price calculation can slow down the subgraph by a lot.

An example could be:
```
type VolumeByToken @entity {
  token: Token!
  volume: BigInt! # In token's native amount
}

cumulativeSaleVolumeByToken: [VolumeByToken]!
```
By doing this, we separate the volume by the denominated token and do USD conversion offline.